src/rust/lib/graph-editor/src/component/visualization.rs

```diff
@@ -0,0 +1,332 @@
+    // TODO remove dummy functionality and use an actual visualisation
+    pub fn set_data(&self, data:Data){
+                self.content.dom().set_inner_html(
+                    &format!(r#"
+<svg>
+  <circle style="fill: #69b3a2" stroke="black" cx=50 cy=50 r={}></circle>
+</svg>
+"#, data.as_json()));
+     }
+
+    /// Make the visualisation visible in the scene.
+    pub fn show(&self) {
+        // TODO investigate how to do this through display_object functionality.
+        self.content.dom().set_style_or_panic("visibility", "visible");
+    }
+
+    /// Hide the visualisation in the scene.
+    pub fn hide(&self) {
+        // TODO investigate how to do this through display_object functionality.
+        self.content.dom().set_style_or_panic("visibility", "hidden");
+    }
+}
+
+impl From<DomSymbol> for Visualization {
+    fn from(symbol: DomSymbol) -> Self {
+        Visualization { content : Rc::new(symbol) }
+    }
+}
+
+impl From<Rc<DomSymbol>> for Visualization {
+    fn from(symbol: Rc<DomSymbol>) -> Self {
+        Visualization { content : symbol }
+    }
+}
+
+
+
+// ============================
+// === Visualization Events ===
+// ============================
+
+/// Visualization events.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Events {
+    pub network           : frp::Network,
+    pub show              : frp::Source,
+    pub hide              : frp::Source,
+    pub toggle_visibility : frp::Source,
+    pub update_visualization    : frp::Source<Option<Visualization>>,
+    pub update_data       : frp::Source<Data>,
+}
+
+impl Default for Events {
+    fn default() -> Self {
+        frp::new_network! { visualization_events
+            def show              = source::<()>                    ();
+            def hide              = source::<()>                    ();
+            def toggle_visibility = source::<()>                    ();
+            def update_visualization    = source::<Option<Visualization>> ();
+            def update_data       = source::<Data>                  ();
+        };
+        let network = visualization_events;
+        Self {network,show,hide,update_visualization,toggle_visibility,update_data}
+    }
+}
+
+
+
+// ================================
+// === Visualizations Container ===
+// ================================
+
+/// Container that wraps a `Visualisation` for rendering and interaction in the GUI.
+#[derive(Clone,CloneRef,Debug)]
+#[allow(missing_docs)]
+pub struct Container {
+    pub data : Rc<ContainerData>
+}
+
+/// Weak version of `Container`.
+#[derive(Clone,CloneRef,Debug)]
+pub struct WeakContainer {
+    data : Weak<ContainerData>
+}
+
+/// Internal data of a `Container`.
+#[derive(Debug,Clone)]
+#[allow(missing_docs)]
+pub struct ContainerData {
+    pub logger : Logger,
+    pub events : Events,
+
+    node          : display::object::Instance,
+    size          : Cell<Vector2<f32>>,
+    position      : Cell<Vector3<f32>>,
```

Comment:
You dont need to store position here, the display object stores it for you and gives you methods like `position` and `mod_position` as soon as you implement `display::Object for ContainerData`.